distributed/http/worker/prometheus/core.py

```diff
@@ -52,6 +52,12 @@ def collect(self):
             value=self.server.latency,
         )
 
+        yield GaugeMetricFamily(
+            self.build_name("comm_reserved_bytes"),
+            "Number of bytes currently reserved for incoming/outgoing data transfers.",
```

Comment:
Not quite right. WorkerState.comm_nbytes is exclusively about the *incoming* bytes.
The two are fundamentally different memory-wise:

- incoming bytes are going to cause a spike in unmanaged memory as soon as they arrive, and only when the transfer is complete they will transition to managed memory and become spillable. Also, no amount of pausing will make them stop coming. So they are at a very high risk of making the worker shoot past the `terminate` threshold.
- outgoing bytes are managed memory. There are two notable exceptions:
  - the key is released while the outgoing comms are in progress
  - the key is spilled out while the outgoing comms are in progress. Notice that the act of sending the key out will make it rise to the top of the SpillBuffer LRU, so this will happen only in cases of extreme spill/unspill activity.

Reply:
Thanks for the clarification!

Reply:
I've overhauled this entire section after #6933